src/main/groovy/com/morpheus/scvmm/sync/IpPoolsSync.groovy

```diff
@@ -0,0 +1,307 @@
+                    doSave = true
+                }
+                if(pool.netmask && network.netmask != pool.netmask) {
+                    network.netmask = pool.netmask
+                    doSave = true
+                }
+
+                if(doSave) {
+                    morpheusContext.async.cloud.network.save(network).blockingGet()
+                }
+            }
+
+            if(subnetId && network) {
+                def subnet = network.subnets.find { it ->
+                    it.externalId?.startsWith(subnetId)
+                }
+                if(subnet) {
+                    def doSave = false
+
+                    if(subnet.pool != pool) {
+                        subnet.pool = pool
+                        doSave = true
+                    }
+                    if(pool.gateway && subnet.gateway != pool.gateway) {
+                        subnet.gateway = pool.gateway
+                        doSave = true
+                    }
+                    if(pool.dnsServers?.size() && pool.dnsServers[0] && subnet.dnsPrimary != pool.dnsServers[0]) {
+                        subnet.dnsPrimary = pool.dnsServers[0] ?: null
+                        doSave = true
+                    }
+                    if(pool.dnsServers?.size() > 1 && pool.dnsServers[1] && subnet.dnsSecondary != pool.dnsServers[1]) {
+                        subnet.dnsSecondary = pool.dnsServers[1] ?: null
+                        doSave = true
+                    }
+                    if(pool.netmask && subnet.netmask != pool.netmask) {
+                        subnet.netmask = pool.netmask
+                        doSave = true
+                    }
+
+                    if(doSave) {
+                        morpheusContext.async.networkSubnet.save(subnet).blockingGet()
+                    }
+                }
+            }
+        } catch (e) {
+            log.error("Error in updateNetworkForPool: ${e}", e)
+        }
+    }
+
+    private updateMatchedIpPools(List<SyncTask.UpdateItem<NetworkPool, Map>> updateList, networks, networkMapping) {
+        log.debug("updateMatchedIpPools : ${updateList.size()}")
+
+        try {
+            updateList?.each { updateMap ->
+                NetworkPool existingItem = updateMap.existingItem
+                def masterItem = updateMap.masterItem
+                if (existingItem) {
+                    // Update the range (if needed)
+                    if(masterItem.IPAddressRangeStart && masterItem.IPAddressRangeEnd) {
+                        if(!existingItem.ipRanges) {
+                            def range = new NetworkPoolRange(networkPool: existingItem, startAddress: masterItem.IPAddressRangeStart, endAddress: masterItem.IPAddressRangeEnd, addressCount: (masterItem.TotalAddresses ?: 0).toInteger(), externalId: masterItem.ID)
+                            morpheusContext.async.cloud.network.pool.poolRange.save(range).blockingGet()
+                            existingItem.addToIpRanges(range)
```

Comment:
do this before save to reduce the number of api calls.

Reply:
@ddeyoung changes has been done as suggested